cmd/lakectl/cmd/branch_tools_test.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/treeverse/lakefs/pkg/metastore"
+	"github.com/treeverse/lakefs/pkg/metastore/mock"
+)
+
+const (
+	dbName     = "source"
+	repoName   = "repo"
+	branchName = "branch"
+)
+
+var (
+	validPath   = fmt.Sprintf("s3://%s/%s/path/to/schema", repoName, branchName)
+	invalidPath = fmt.Sprintf("s3/%s/%s/", repoName, branchName)
+)
+
+func TestExtractRepoAndBranchFromDBName(t *testing.T) {
+	type args struct {
+		ctx    context.Context
+		dbName string
+	}
+	tests := []struct {
+		name      string
+		args      args
+		want      string
+		want1     string
+		wantErr   bool
```

Comment:
This test is very brittle.

Can `wantErr` be an `error`, that we can then check with `errors.Is`?  Or, if the is impossible, a substring of the expected error??  Otherwise any scenario with `wantErr == true` (most of them) will happily pass as long as _something_ on the execution path fails.